napari/components/_layers_list/view.py

```diff
@@ -38,46 +39,62 @@ def __init__(self, layers):
         self.drag_name = None
 
     def _add(self, event):
-        """Insert `event.widget` at index `event.index`."""
+        """Insert widget for layer `event.item` at index `event.index`."""
         layer = event.item
-        index = event.index
         total = len(self.layers)
-        if layer._qt_properties is not None:
-            self.vbox_layout.insertWidget(
-                2 * (total - index) - 1, layer._qt_properties
-            )
-            self.vbox_layout.insertWidget(2 * (total - index), QtDivider())
+        index = 2 * (total - event.index) - 1
+        properties = create_qt_properties(layer)
+        self.vbox_layout.insertWidget(index, properties)
+        self.vbox_layout.insertWidget(index + 1, QtDivider())
 
     def _remove(self, event):
-        """Remove layer widget at index `event.index`."""
-        layer = event.item
-        if layer._qt_properties is not None:
-            index = self.vbox_layout.indexOf(layer._qt_properties)
-            divider = self.vbox_layout.itemAt(index + 1).widget()
-            self.vbox_layout.removeWidget(layer._qt_properties)
-            layer._qt_properties.deleteLater()
-            layer._qt_properties = None
-            self.vbox_layout.removeWidget(divider)
-            divider.deleteLater()
-            divider = None
+        """Remove widget for layer at index `event.index`."""
+        layer_index = event.index
+        total = len(self.layers)
+        # Find property widget and divider for layer to be removed
+        index = 2 * (total - layer_index) + 1
+        widget = self.vbox_layout.itemAt(index).widget()
+        divider = self.vbox_layout.itemAt(index + 1).widget()
+        self.vbox_layout.removeWidget(widget)
+        widget.deleteLater()
+        widget = None
+        self.vbox_layout.removeWidget(divider)
+        divider.deleteLater()
+        divider = None
 
     def _reorder(self, event):
         """Reorders list of layer widgets by looping through all
         widgets in list sequentially removing them and inserting
         them into the correct place in final list.
         """
         total = len(self.layers)
+
+        # Create list of the current property and divider widgets
+        widgets = [
+            self.vbox_layout.itemAt(i + 1).widget() for i in range(2 * total)
+        ]
+        indices = [
+            self.layers.index(w.layer)
+            for i, w in enumerate(widgets)
+            if i % 2 == 0
```

Comment:
could you explain why you're only taking out evenly-indexed widgets? does this have to do with ignoring the dividers?

Reply:
yes, have added a comment